src/drivers/sdl/lv_sdl_window.c

```diff
@@ -181,6 +189,14 @@ void lv_sdl_window_set_title(lv_display_t * disp, const char * title)
     SDL_SetWindowTitle(dsc->window, title);
 }
 
+void lv_sdl_window_set_icon(lv_display_t * disp, uint16_t * icon)
+{
+    lv_sdl_window_t * dsc = lv_display_get_driver_data(disp);
```

Comment:
```suggestion
    lv_sdl_window_t * dsc = (lv_sdl_window_t *) lv_display_get_driver_data(disp);
```

So this code stays compatible with c++

Reply:
Thank you, we will update this line.

Reply:
We only do these casts in the examples, AFAIK. There are a lot of places all over LVGL where implicit `void` pointer coercion happens

Reply:
Currently, yes, i believe you're right but wouldn't you agree that it's a good idea to start adding these ?

Reply:
Personally, no! :laughing: is there a strong case for it?

Reply:
Fair point ahaha

1. Explicitness and readability, it tells the rest of the world you know what you're expecting from the function
2. Portability. The code simply doesn't compile with a c++ compiler 

```c
typedef struct {
	int x;
} bar;
bar a;
void *foo(void)
{
	return &a;
}
int main(void)
{
	bar *b = foo();

	b->x = 10;
}
```

```bash
> gcc -Wall -Wextra -pedantic -o main main.c
> g++ -Wall -Wextra -pedantic -o main main.c
main.c: In function ‘int main()’:
main.c:16:21: error: invalid conversion from ‘void*’ to ‘bar*’ [-fpermissive]
   16 |         bar *b = foo();
      |                  ~~~^~
      |                     |
      |                     void*
```

3. Maintainability : 

This compiles:

```c
typedef struct {
	int x;
} bar;

typedef struct {
	float x;
} baz;

bar a;

void *foo(void)
{
	return &a;
}

int main(void)
{
	baz *b;

	b = foo();

	b->x = 10;
}
```

This doesn't compile

```c
typedef struct {
	int x;
} bar;

typedef struct {
	float x;
} baz;

bar a;

void *foo(void)
{
	return &a;
}

int main(void)
{
	baz *b;

	b = (bar *)foo();

	b->x = 10;
}
```
```bash
> gcc -Wall -Wextra -pedantic -o main main.c
main.c: In function ‘main’:
main.c:22:11: error: assignment to ‘baz *’ from incompatible pointer type ‘bar *’ [-Wincompatible-pointer-types]
   22 |         b = (bar *)foo();
      |           ^
```

Reply:
For (1.), why would someone compile a C file as C++? The examples are copy-pasted sometimes so it makes sense, but why for anything else?

I think (2.) is a 50/50 chance whether they put the correct cast. What if they do `b = (baz *)foo();`? In any case it's 50% better than nothing so you've convinced me, somewhat.

Reply:
1) Well, since there are some C and Cpp files in lvgl the simplest way to compile and app using lvgl could be collecting every .c, .cpp and .S file using `find` and then compiling everything using a single command for instance :

```bash
g++ $SRC_FILES -o app
```
Since there are .cpp files inside the source tree, the only way this can possible work is with a c++ compiler but if the C code isn't C++ compatible you remove that option from people

Of course you can always decide that you don't want to support building lvgl that way which is totally acceptable aswell

2) i don't think the problem is using the correct cast, in general the first time the code is written, the variable type (and thus the cast) are correct and you won't have any problems. The cast will save you in future updates, when for some reason you update the type of `b` and don't realize that an implicit cast to the previous type exists

Reply:
1. I see the convenience of being able to do that, but C is more than a subset. It's its own language with divergent semantics.

2. Ah true it will help you if `b` changes, which is worth it enough. But not for the return value of `foo`. I guess the refactorer should check these users anyways.

Reply:
This line will not compile in C++ because `new` is a reserved keyword.

https://github.com/lvgl/lvgl/blob/b7061642a0c78e9f92b0cbabfb7215ba797a1c6f/src/stdlib/lv_mem.c#L141

Reply:
> 1. I see the convenience of being able to do that, but C is more than a subset. It's its own language with divergent semantics.

True but I don't think there's anything in the lvgl source code folder that would make it impossible to be C++ compliant. Most of the changes simply rely on an explicit cast or simple variable rename

> This line will not compile in C++ because `new` is a reserved keyword.

Yep, good one

This command will give you the whole list of things non-C++ compliant. Spoiler alert, there's a lot of them

```bash
g++ -c $(find src -type f \( -name '*.S' -o -name '*.cpp' -o -name '*.c' \))
```

Reply:
Hello AndreCostaaa, liamHowatt,  Thanks for your comments!  May I know the conclusion for this suggest change?

Reply:
Since there is no agreement on the c++ compatible update, can we update the PR to address other review comments first?

Reply:
Yes, feel free to ignore this suggested change in your PR update

Thank you for your work and sorry for hijacking your PR for discussing lvgl c++ compliance :)

Reply:
Thanks for your review and comments.